tarofreighter/chain_porter.go

```diff
@@ -649,54 +609,21 @@ func (p *ChainPorter) stateStep(currentPkg sendPackage) (*sendPackage, error) {
 		ctx, cancel := p.WithCtxQuitNoTimeout()
 		defer cancel()
 
-		// We'll validate the selected input and commitment. From this
-		// we'll gain the asset that we'll use as an input and info
-		// w.r.t if we need to use an unspendable zero-value root.
-		inputAsset, fullValue, err := taroscript.IsValidInput(
-			currentPkg.InputAsset.Commitment, *currentPkg.ReceiverAddr,
-			*currentPkg.InputAsset.Asset.ScriptKey.PubKey,
-			*p.cfg.ChainParams,
-		)
-		if err != nil {
-			return nil, err
-		}
-
-		currentPkg.SendDelta.InputAssets[currentPkg.InputAssetPrevID] = inputAsset
-
 		// Before we can prepare output assets for our send, we need to
 		// generate a new internal key and script key. The script key
 		// is needed for asset change, and the internal key will anchor
 		// the send itself.
 		//
 		// TODO(jhb): ScriptKey derivation instructions should be
 		// specified in the AssetParcel
+		var err error
```

Comment:
Already looking much nicer in this diff!

Still getting through, but I think the extra step here would then be moving a lot of the state like `SenderNewInternalKey` into something that better encapsulates all the data needed to progress with the send.

Reply:
Yes, that's done in later commits.